gensim/models/experimental/drmm_tks.py

```diff
@@ -0,0 +1,595 @@
+                else:
+                    # If the key isn't there give it the zero word index
+                    translated_sentence.append(self.unk_word_index)
+                    n_skipped_words += 1
+            if len(sentence) > self.text_maxlen:
+                logger.info("text_maxlen: %d isn't big enough. Error at sentence of length %d. Sentence is %s" % (
+                    self.text_maxlen, len(sentence), str(sentence)))
+            translated_sentence = translated_sentence + \
+                (self.text_maxlen - len(sentence)) * [self.pad_word_index]
+            translated_data.append(np.array(translated_sentence))
+
+        logger.info("Found %d unknown words. Set them to unknown word index : %d" %
+                    (n_skipped_words, self.unk_word_index))
+        return np.array(translated_data)
+
+    def predict(self, queries, docs):
+        """Predcits on the input paramters using the trained model
+
+        Parameters:
+        -----------
+        queries: list of list of string words
+            The questions for the similarity learning model
+
+            Example:
+            queries=["When was World Wat 1 fought ?".split(),
+                     "When was Gandhi born ?".split()],
+
+        docs: list of list of list of string words
+            The candidate answers for the similarity learning model
+
+            Example:
+            docs = [
+                    ["The world war was bad".split(),
+                    "It was fought in 1996".split()],
+                    ["Gandhi was born in the 18th century".split(),
+                     "He fought for the Indian freedom movement".split(),
+                     "Gandhi was assasinated".split()]
+                   ]
+        """
+        doc_lens = []
+        long_doc_list = []
+        for doc in docs:
+            long_doc_list.append(doc)
+            doc_lens.append(len(doc))
+
+        long_queries = []
+        for doc_len, q in zip(doc_lens, queries):
+            for i in range(len(docs)):
+                long_queries.append(q)
+
+        indexed_long_queries = self.translate_user_data(long_queries)
+        indexed_long_doc_list = self.translate_user_data(long_doc_list)
+        return self.model.predict(
+            x={'query': indexed_long_queries, 'doc': indexed_long_doc_list})
+
+    def save(self, fname, *args, **kwargs):
+        """Save the model. This saved model can be loaded again using :func:`~gensim.models.word2vec.Word2Vec.load`,
+        which supports online training and getting vectors for vocabulary words.
+
+        Parameters
+        ----------
+        fname : str
+            Path to the file.
+
+        """
+        # don't save the keras model as it needs to be saved with a keras function
+        kwargs['ignore'] = kwargs.get('ignore', ['model'])
+        kwargs['fname_or_handle'] = fname
+        super(DRMM_TKS, self).save(*args, **kwargs)
+        self.model.save(fname + ".keras")
+
+
+class _drmm_tks:
```

Comment:
You really no need this class, you can integrate this code to `DRMM_TKS` freely.